Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.cpp

```diff
@@ -514,25 +514,23 @@ void GraphicsContextGLANGLE::readnPixelsImpl(GCGLint x, GCGLint y, GCGLsizei wid
 
 void GraphicsContextGLANGLE::validateDepthStencil(ASCIILiteral packedDepthStencilExtension)
 {
-    // FIXME: Since the constructors of various platforms are not shared, we initialize this here.
-    // Upon constructing the context, always initialize the extensions that the WebGLRenderingContext* will
-    // use to turn on feature flags.
-    String packedDepthStencilExtensionString { packedDepthStencilExtension };
-    if (supportsExtension(packedDepthStencilExtensionString)) {
-        ensureExtensionEnabled(packedDepthStencilExtensionString);
-        m_internalDepthStencilFormat = GL_DEPTH24_STENCIL8_OES;
-    } else
-        m_internalDepthStencilFormat = GL_DEPTH_COMPONENT16;
-
     auto attrs = contextAttributes();
-    if (attrs.stencil) {
-        if (m_internalDepthStencilFormat == GL_DEPTH24_STENCIL8_OES) {
-            // Force depth if stencil is true.
-            attrs.depth = true;
-        } else
+    if (attrs.stencil && attrs.depth) {
+        ASSERT(packedDepthStencilExtension == "GL_OES_packed_depth_stencil"_s);
+        String packedDepthStencilExtensionString { packedDepthStencilExtension };
+        if (supportsExtension(packedDepthStencilExtensionString)) {
+            // This extension is always enabled when supported
+            m_internalDepthStencilFormat = GL_DEPTH24_STENCIL8_OES;
+        } else {
+            // Combined buffer not supported, prefer depth when both requested
+            m_internalDepthStencilFormat = GL_DEPTH_COMPONENT16;
```

Comment:
Would GL_DEPTH_COMPONENT24_OES be a good choice if GL_OES_depth24 is supported? Applications tend to notice the depth buffer precision differences between 16-bit and 24-bit depth buffers.

Reply:
Done. FWIW, depth buffer is always 32F when ANGLE/Metal is used.